java.lang.NullPointerException: Attempt to invoke virtual method 'io.flutter.plugin.common.PluginRegistry$RequestPermissionsResultListener com.lyokone.location.FlutterLocationService.getServiceRequestPermissionsResultListener()' on a null object reference

**Describe the bug**
App crashing on very first app open

**Tested on:**
 - Android OS 13, Samsung Galaxy M13

**Other plugins:**
 - List of others Flutter plugins that could interfere


**Additional logs**
D/FlutterGeolocator(20666): Attaching Geolocator to activity
D/FlutterGeolocator(20666): Detaching Geolocator from activity
E/AndroidRuntime(20666): FATAL EXCEPTION: main
E/AndroidRuntime(20666): Process: com.skieats.owner, PID: 20666
E/AndroidRuntime(20666): java.lang.RuntimeException: Unable to destroy activity {com.skieats.owner/com.skieats.owner.MainActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'io.flutter.plugin.common.PluginRegistry$RequestPermissionsResultListener com.lyokone.location.FlutterLocationService.getServiceRequestPermissionsResultListener()' on a null object reference
E/AndroidRuntime(20666): 	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:6080)
E/AndroidRuntime(20666): 	at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:6113)
E/AndroidRuntime(20666): 	at android.app.ActivityThread.handleRelaunchActivityInner(ActivityThread.java:6460)
E/AndroidRuntime(20666): 	at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:6341)
E/AndroidRuntime(20666): 	at android.app.servertransaction.ActivityRelaunchItem.execute(ActivityRelaunchItem.java:71)
E/AndroidRuntime(20666): 	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
E/AndroidRuntime(20666): 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
E/AndroidRuntime(20666): 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
E/AndroidRuntime(20666): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2574)
E/AndroidRuntime(20666): 	at android.os.Handler.dispatchMessage(Handler.java:106)
E/AndroidRuntime(20666): 	at android.os.Looper.loopOnce(Looper.java:226)
E/AndroidRuntime(20666): 	at android.os.Looper.loop(Looper.java:313)
E/AndroidRuntime(20666): 	at android.app.ActivityThread.main(ActivityThread.java:8757)
E/AndroidRuntime(20666): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(20666): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
E/AndroidRuntime(20666): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
E/AndroidRuntime(20666): Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'io.flutter.plugin.common.PluginRegistry$RequestPermissionsResultListener com.lyokone.location.FlutterLocationService.getServiceRequestPermissionsResultListener()' on a null object reference
E/AndroidRuntime(20666): 	at com.lyokone.location.LocationPlugin.dispose(LocationPlugin.java:117)
E/AndroidRuntime(20666): 	at com.lyokone.location.LocationPlugin.detachActivity(LocationPlugin.java:56)

--- a/packages/location/android/src/main/java/com/lyokone/location/LocationPlugin.java
+++ b/packages/location/android/src/main/java/com/lyokone/location/LocationPlugin.java
@@ -84,7 +84,9 @@ public class LocationPlugin implements FlutterPlugin, ActivityAware {
         @Override
         public void onServiceConnected(ComponentName name, IBinder service) {
             Log.d(TAG, "Service connected: " + name);
-            initialize(((FlutterLocationService.LocalBinder) service).getService());
+            if(service instanceof FlutterLocationService.LocalBinder){
+                initialize(((FlutterLocationService.LocalBinder) service).getService());
+            }
         }
 
         @Override
@@ -114,12 +116,14 @@ public class LocationPlugin implements FlutterPlugin, ActivityAware {
         methodCallHandler.setLocationService(null);
         methodCallHandler.setLocation(null);
 
-        activityBinding.removeRequestPermissionsResultListener(locationService.getServiceRequestPermissionsResultListener());
-        activityBinding.removeRequestPermissionsResultListener(locationService.getLocationRequestPermissionsResultListener());
-        activityBinding.removeActivityResultListener(locationService.getLocationActivityResultListener());
+        if(locationService != null){
+            activityBinding.removeRequestPermissionsResultListener(locationService.getServiceRequestPermissionsResultListener());
+            activityBinding.removeRequestPermissionsResultListener(locationService.getLocationRequestPermissionsResultListener());
+            activityBinding.removeActivityResultListener(locationService.getLocationActivityResultListener());
 
-        locationService.setActivity(null);
+            locationService.setActivity(null);
 
-        locationService = null;
+            locationService = null;
+        }
     }
 }

